components/model/category/EditCategoryForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { css, useTheme } from '@emotion/react';
+import { FormEventHandler, MouseEventHandler, useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { updateCategory } from '../../../data/actions';
+import { useCategoryList } from '../../../data/hooks';
+import { HttpError } from '../../../data/libs/fetchers';
+import { userTokenState } from '../../../states';
+import { Icon } from '../../ui/data-display/Icon';
+import { Spinner } from '../../ui/feedback/Spinner';
+import { Button } from '../../ui/input/Button';
+import { TextInput } from '../../ui/input/TextInput';
+
+/** @jsxImportSource @emotion/react */
+interface Props {
+  categoryId: string;
+  onCancel?: () => void;
+  onSubmitted?: () => void;
+}
+
+const labelStyle = css`
+  font-size: 14px;
+  font-weight: 500;
+`;
+
+export const EditCategoryForm = ({ categoryId, onCancel: emitCancelEvent, onSubmitted }: Props) => {
+  const theme = useTheme();
+  const { categories, isLoading } = useCategoryList();
+  const userToken = useRecoilValue(userTokenState);
+
+  const [categoryName, setCategoryName] = useState('');
+
+  useEffect(() => {
+    if (!categories) {
+      return;
+    }
+
+    const category = categories.filter((category) => category.id === categoryId)[0];
+    setCategoryName(category.name);
+  }, [categories, categoryId, isLoading]);
+
+  const onSubmit: FormEventHandler<HTMLFormElement> = async (e) => {
+    e.preventDefault();
+
+    try {
+      await updateCategory({ categoryId, token: userToken, name: categoryName });
+      onSubmitted && onSubmitted();
+    } catch (e) {
+      if (e instanceof HttpError) {
+        window.alert('更新に失敗しました。Error Message: ' + e.data.error);
+        return;
+      }
+      throw e;
+    }
+  };
+
+  const onCancel: MouseEventHandler<HTMLButtonElement> = (e) => {
+    e.preventDefault();
+    emitCancelEvent && emitCancelEvent();
```

Comment:
fyi
既知でしたらすみません、こんな書き方もできたりします
```suggestion
    emitCancelEvent?.()
```

Reply:
それは知りませんでした！関数もオプショナル構文でかけるんですね。
1183efb593f4e93379dae9bb955c285fb0212156
にて修正しました。他にも同様の箇所ありますが、次回からその書き方でしようと思います。